src/nsolid/nsolid_heap_snapshot.cc

```diff
@@ -12,6 +14,59 @@ NSolidHeapSnapshot::NSolidHeapSnapshot() {
   ASSERT_EQ(0, in_progress_heap_snapshots_.init(true));
 }
 
+int NSolidHeapSnapshot::StartTrackingHeapObjects(
+    SharedEnvInst envinst,
+    bool redacted,
+    bool trackAllocations,
+    uint64_t duration,
+    void* data,
+    Snapshot::snapshot_proxy_sig proxy,
+    internal::deleter_sig deleter) {
+  uint64_t thread_id = envinst->thread_id();
+  nsuv::ns_mutex::scoped_lock lock(&in_progress_heap_snapshots_);
+  // We can not trigger this command if there is already a snapshot in progress
+  auto it = threads_running_snapshots_.find(thread_id);
+  if (it != threads_running_snapshots_.end()) {
```

Comment:
Are we using the same map because we're not allowing taking a heap profile while taking a snapshot and viceversa. If that's the case, we should document it, otherwise we should be using different map/lock

Reply:
When you start the tracking heap objects, we consider that as a “running/in progress” heap snapshot, that’s why I include that in the map/lock.